src/workerd/api/r2-bucket.c++

```diff
@@ -274,7 +314,7 @@ void initGetOptions(jsg::Lock& js, Builder& builder, Options& o) {
   }
 }
 
-static bool isQuotedEtag(kj::StringPtr etag) {
+static bool isValidEtag(kj::StringPtr etag) {
   return etag.startsWith("\"") && etag.endsWith("\"");
```

Comment:
At some point, it would likely be a good idea to properly validate the Etag content matches the spec'd definition:

```
ETag       = entity-tag

entity-tag = [ weak ] opaque-tag
weak       = %x57.2F ; "W/", case-sensitive
opaque-tag = DQUOTE *etagc DQUOTE
etagc      = %x21 / %x23-7E / obs-text
           ; VCHAR except double quotes, plus obs-text
```

Reply:
I did base my implementation of parseConditionalEtagHeader on this http spec. I actually need to just remove this call to isValidEtag entirely for this codepath that handles headers being passed since parseConditionalEtagHeader already does the required checks surrounding formatting.

I think my implementation is slightly more lax than the http spec requires me to be. For example, I accept etag headers like if-none-match: "example-etag", *, "another etag", and just turn that into a wildcard match. I don't think that should be a problem, but I can try to change my implementation to be more strict.